### PR TITLE
Geofeat to isochrone microservice

### PR DESCRIPTION
**Pull request by punkerpunker**
I've decided to remove everything that is out of isochrones scope out of this project. Generally speaking, isochrones itself is a core functionality of `geofeat`, everything else was a burden. 

Seems like it is easier just to walk through the new branch in opposite to compare it with master because project became completely different.

### CODE REVIEW

**Review by temsiPatrin — APPROVED** [review]

File: containers/app/app.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            os.makedirs(directory)
+
+    @staticmethod
+    def convert(raw_directory, converted_directory):
+        for file in os.listdir(raw_directory):
+            if file.endswith(".osm.pbf"):
+                path = os.path.join(raw_directory, file)
+                osm_name = file.rstrip('.pbf')
+                new_path = os.path.join(converted_directory, osm_name)
+                subprocess.check_call([f'osmconvert {path} --drop-author --drop-version --out-osm -o={new_path}'], shell=True)
```
Comment (MTKarenin) [inline containers/app/app.py]: Again about the naming.
This name does not reveal what function do. User doesn't understand what is raw file and what is conwerted file.
I suggest: 
```suggestion
    @staticmethod
    def convert_pbf_to_osm(pbf_file_directory: str, osm_file_directory: str):
        for file in os.listdir(pbf_file_directory):
            if file.endswith(".osm.pbf"):
                path = os.path.join(pbf_file_directory, file)
                osm_name = file.rstrip('.pbf')
                new_path = os.path.join(osm_file_directory, osm_name)
                subprocess.check_call([f'osmconvert {path} --drop-author --drop-version --out-osm -o={new_path}'], shell=True)
```

File: containers/app/app.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+CONVERTED_DIR = '/osm/pbf/converted'
+MAPCONFIG_PATH = '/osm/mapconfig.xml'
+engine = sqlalchemy.create_engine(f'postgresql+psycopg2://postgres:{DB_PASSWORD}@pgrouting/postgres')
+
+
+class Loader:
+    @staticmethod
+    def create_dirs(directory):
+        if not os.path.exists(directory):
+            os.makedirs(directory)
```
Comment (MTKarenin) [inline containers/app/app.py]: There is an inconsisency between function name and what it does. Accordin to it's name this function should create multiple directories, but it creates only one directory. I suggest to rename this function/
```suggestion
    @staticmethod
    def create_directory(path_to_directory: str):
        if not os.path.exists(path_to_directory):
            os.makedirs(path_to_directory)
```

File: containers/app/app.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                osm_name = file.rstrip('.pbf')
+                new_path = os.path.join(converted_directory, osm_name)
+                subprocess.check_call([f'osmconvert {path} --drop-author --drop-version --out-osm -o={new_path}'], shell=True)
+
+    @staticmethod
+    def load(converted_directory, mapconfig_path, password):
+        for file in os.listdir(converted_directory):
+            path = os.path.join(converted_directory, file)
+            subprocess.check_call([f'osm2pgrouting --host pgrouting --f {path} --conf {mapconfig_path} --dbname postgres '
+                                   f'--username postgres --password {password}'], shell=True)
```
Comment (MTKarenin) [inline containers/app/app.py]: And again - you don't realy understand what does this function load exactly
```suggestion
    @staticmethod
    def load_osm_files_to_db(osm_file_directory: str, mapconfig_path: str, password: str):
        for file in os.listdir(osm_file_directory):
            path = os.path.join(osm_file_directory, file)
            subprocess.check_call([f'osm2pgrouting --host pgrouting --f {path} --conf {mapconfig_path} --dbname postgres '
                                   f'--username postgres --password {password}'], shell=True)
```

File: containers/app/app.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    with engine.connect() as connection:
+        result = connection.execute(query)
+        for row in result:
+            return row['polygon']
+
+
+if __name__ == "__main__":
+    Loader.create_dirs(CONVERTED_DIR)
+    Loader.convert(RAW_DIR, CONVERTED_DIR)
+    Loader.load(CONVERTED_DIR, MAPCONFIG_PATH, DB_PASSWORD)
```
Comment (MTKarenin) [inline containers/app/app.py]:
```suggestion
    Loader.create_directory(OSM_DIR)
    Loader.convert_pbf_to_osm(PBF_DIR, OSM_DIR)
    Loader.load_osm_files_to_db(OSM_DIR, MAPCONFIG_PATH, DB_PASSWORD)
```

File: containers/app/app.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import connexion
+import sqlalchemy
+from shapely.geometry import Polygon
+from shapely import wkt
+
+
+DB_PASSWORD = os.environ['POSTGRES_PASSWORD']
+RAW_DIR = '/osm/pbf'
+CONVERTED_DIR = '/osm/pbf/converted'
+MAPCONFIG_PATH = '/osm/mapconfig.xml'
```
Comment (MTKarenin) [inline containers/app/app.py]: See comments below
```suggestion
PBF_DIR = '/osm/pbf'
OSM_DIR = '/osm/pbf/converted'
MAPCONFIG_PATH = '/osm/mapconfig.xml'
```

**Review by MTKarenin — CHANGES_REQUESTED** [review]

File: containers/app/app.py (hunk 2 above)
Comment (AlexKay28) [reply containers/app/app.py]: os.makedirs() method in Python is used to create a directory recursively. Actually, it looks good, but I agree that name should be more clear in terms of taken goal

File: docker-compose.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+services:
+  pgrouting:
+    container_name: pgrouting
+    restart: always
+    image: pgrouting/pgrouting:13-3.0-3.1.0
+    volumes:
+      - ./containers/pgrouting/data:/var/lib/postgresql/data
+      - ./containers/pgrouting/init.sql:/docker-entrypoint-initdb.d/init.sql
+    environment:
+      - POSTGRES_PASSWORD=tttBBB777
```
Comment (AlexKay28) [inline docker-compose.yaml]: maybe it would be right to change this pass to something else?)

**Review by AlexKay28 — APPROVED** [review]

**Pull request merged**